Add getDynamicValue for SizeMul and SizeDiv nodes.

vanbasten23 opened the pull request.

JackCaoG reviewed (approved):
Thanks!

miladm commented on torch_xla/csrc/ops/dynamic_ir.cpp:
```diff
@@ -121,6 +129,14 @@ SizeDiv::SizeDiv(torch::lazy::Value a, torch::lazy::Value b)
+  const torch::lazy::DimensionNode* dim_node_1 = DimCast(operand(1));
+  XLA_CHECK(dim_node_0);
+  XLA_CHECK(dim_node_1);
+  return dim_node_0->getDynamicValue() / dim_node_1->getDynamicValue();
```
Why do we not have the following check?

```
XLA_CHECK(dim_node_1->getDynamicValue() != 0)
      << "Can't divide a dimension by zero";
```

CC @vanbasten23 @JackCaoG

JackCaoG replied:
yea, I guess we can have this check

miladm replied:
@vanbasten23 can you please add the PR?

vanbasten23 replied:
Thanks for pointing it out. Working on it.

miladm commented on test/cpp/test_ir.cpp:
```diff
@@ -279,5 +306,30 @@ TEST_F(IrTest, TestSizeDivNode) {
+  std::shared_ptr<torch::lazy::DimensionNode> dim_node_div =
+      std::dynamic_pointer_cast<torch::lazy::DimensionNode>(node_div);
+  EXPECT_EQ(dim_node_div->getStaticValue(), 50);
+  EXPECT_EQ(dim_node_div->getDynamicValue(), 0);
```
This test doesn't check for divide by zero. How do we address that?

@vanbasten23

vanbasten23 replied:
Created a test: https://github.com/pytorch/xla/pull/4051

miladm reviewed (commented):
Reviewed this PR. Thanks @vanbasten23. 

I added a couple of comments to the PR. Since it's already merged, feel free to open a new PR to address comments.

vanbasten23 commented:
> Reviewed this PR. Thanks @vanbasten23.
> 
> I added a couple of comments to the PR. Since it's already merged, feel free to open a new PR to address comments.

Thanks for reviewing. Addressed it in https://github.com/pytorch/xla/pull/4051.
